### ISSUE TITLE
Babel plugin does not cover all type-infer situations 

### ISSUE THREAD

**Issue by XHMM** [realm/realm-js#5111]
### How frequently does the bug occur?

All the time

### Description

Demo code when working with @realm/babel-plugin@0.1.0 and realm@11.1.0 :
```js
import Realm from "realm";

export class NoteSchema extends Realm.Object<NoteSchema, "content"> {
  // @ts-ignore
  static name = "MyNote"
  _id = new Realm.BSON.ObjectId()
  createdAt = Date.now()
  content!: string
}
```

After babel transformation, **`createdAt` was missed in schema**:
```js
NoteSchema.schema = {
    name: "NoteSchema",
    properties: {
      _id: {
        type: "objectId",
        default: function _default() {
          return new _realm.default.BSON.ObjectId();
        }
      },
      content: {
        type: "string"
      }
    },
    name: "MyNote"
  };
```

### Can you reproduce the bug?

Yes, always

### Version

@realm/babel-plugin@0.1.0 and realm@11.1.0

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

it's just a js side issue

**Comment by kneth** [comment]
@XHMM Thank you for reporting. Do you also observe it if you only have one function (yeah, I am mentally debugging already)?

**Comment by XHMM** [comment]
Did you mean this code?
```js
export class NoteSchema extends Realm.Object<NoteSchema> {
  createdAt = Date.now()
}
```

The above code still does not work, and the console logged "transform[stderr]: Unable to determine type of 'createdAt' property"

**Comment by kneth** [comment]
[`Date.now()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now) returns a number (milliseconds after epoc) while [`new Date()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) returns a `Date` object as we expect.

**Comment by XHMM** [comment]
You're right, but here what I want is to store the timestamp(number) but not date. (sorry for the field name mislead you)

Also I tried write `createdAt = new Date()` and babel transformation was right, but `createdAt = Date.now()` didn't transform correctly as the issue showed.

**Comment by takameyer** [comment]
Hi all, I've been updating some of our templates to use the new babel plugin and I also ran into this issue.  The solution is to explicitly declare the types in your class and not infer them.  The babel plugin doesn't seem to be able to retrieve the inferred type (perhaps we can improve this somehow).  

```
export class NoteSchema extends Realm.Object<NoteSchema, "content"> {
  // @ts-ignore
  static name = "MyNote"
  _id: Realm.BSON.ObjectId = new Realm.BSON.ObjectId()  // Set Realm.BSON.ObjectId explicitly
  createdAt: Date = Date.now() // Set Date explicitly
  content!: string
}
```

Let us know if that helps!

**Comment by XHMM** [comment]
> Hi all, I've been updating some of our templates to use the new babel plugin and I also ran into this issue. The solution is to explicitly declare the types in your class and not infer them. The babel plugin doesn't seem to be able to retrieve the inferred type (perhaps we can improve this somehow).
> 
> ```ts
> export class NoteSchema extends Realm.Object<NoteSchema, "content"> {
>   // @ts-ignore
>   static name = "MyNote"
>   _id: Realm.BSON.ObjectId = new Realm.BSON.ObjectId()  // Set Realm.BSON.ObjectId explicitly
>   createdAt: Realm.Types.Int = Date.now() // Set to Int explicitly
>   content!: string
> }
> ```
> 
> Let us know if that helps!

Yes this way works, and waiting for a fix so we can fully leverage typescript types without worry😊
